ui/pages/confirmations/hooks/useQueuedConfirmationEvents.ts

```diff
@@ -0,0 +1,35 @@
+import { useContext, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+  QueueType,
+} from '../../../../shared/constants/metametrics';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+import {
+  getQueuedRequestCount,
+  pendingApprovalsSortedSelector,
+} from '../../../selectors';
+
+export const useQueuedConfirmationsEvent = (queueType: QueueType) => {
+  const pendingApprovals = useSelector(pendingApprovalsSortedSelector);
+  const queuedRequestCount = useSelector(getQueuedRequestCount);
+  const trackEvent = useContext(MetaMetricsContext);
+
+  const properties = {
+    confirmation_type: pendingApprovals[0].type,
+    referrer: pendingApprovals[0].origin,
+    queue_size: queuedRequestCount,
+    queue_type: queueType,
+  };
+
+  useEffect(() => {
+    if (queuedRequestCount > 0) {
+      trackEvent({
+        event: MetaMetricsEventName.ConfirmationQueued,
+        category: MetaMetricsEventCategory.Confirmations,
+        properties,
+      });
+    }
+  }, [JSON.stringify(properties), queuedRequestCount]);
+};
```

Comment:
It looks bit unusual to have hook which does not return anything but itself performs task, an option could be we return a callback from this hook which user can invoke.

Reply:
It's rare, but it feels like a good usage since we can automate the metrics based on dependencies and encapsulate that logic, rather than it being a component problem.

We do something similar in `useSimulationMetrics` already.

Concerning the documentation, it feels suitable since we are still synchronizing with an external system, in this case Segment.